recipe/meta.yaml

```
@@ -1,80 +1,133 @@
-    - '*/libtorch_python.so'
+  number: 0
+  skip: true  # [cuda_compiler_version == "None"]
+  skip: true  # [osx or win]
+  skip: true  # [py == 38]
 
+# break CI jobs per python version
 requirements:
-  
-  build:
-    - cmake
-    - make
-    - {{ compiler('c') }}
-    - {{ stdlib("c") }}
-    - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}
-
   host:
     - python
-    - packaging
-    - setuptools >=49.4.0
-    - pytorch >=2.1.0
-    - wheel
-    - pip
-    - libxcrypt
-  run:
-    - python
-    - pytorch
-    - torchvision
-    - transformers >=4.35.0
-    - tokenizers >=0.12.1
-    - accelerate
-    - sentencepiece
-    - lm_eval
-    - texttable
-    - toml
-    - attributedict
-    - protobuf
-    - torchvision
-    - tabulate
-    - cudatoolkit
-    - libtorch
 
-test:
-  imports:
-    - awq
-  # commands: # pending https://github.com/conda-forge/tcolorpy-feedstock/issues/12
-  #   - pip check
-  requires:
-    - pip
+outputs:
+  - name: autoawq_kernels
+    version: {{ kernel_version }}
+    build:
+      activate_in_script: true
+      script:
+        - sed -i 's/os.environ\["CC"\] = "g++"//' kernels/setup.py
+        - sed -i 's/os.environ\["CXX"\] = "g++"//' kernels/setup.py
+        - cd kernels
+        - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+      skip: true  # [cuda_compiler_version == "None"]
+      skip: true  # [osx or win]
+      skip: true  # [py == 38]
+      missing_dso_whitelist:
+        - '*/libtorch_python.so'
+
+    requirements:
+      build:
+        - cmake
+        - make
+        - {{ stdlib('c') }}
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('cuda') }}
+      host:
+        - python
+        - pip
+        - pytorch
+        {% if cuda_major >= 12 %}
+        - cuda-version {{ cuda_compiler_version }}
+        - cuda-cudart-dev
+        - libcublas-dev
+        - libcusolver-dev
+        - libcusparse-dev
+        {% endif %}
+    test:
+      imports:
+        - awq_ext
+
+{% if False %}
+  - name: autoawq
+    build:
+      script:
+        - export LDFLAGS="$(echo $LDFLAGS | sed 's/-Wl,--allow-shlib-undefined//g')"
+        - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+      skip: true  # [cuda_compiler_version == "None"]
+      skip: true  # [osx or win]
+      skip: true  # [py == 38]
+    requirements:
+      build:
+        - {{ stdlib('c') }}
+        - {{ compiler('c') }}
+      host:
+        - python
+        - {{ pin_subpackage('autoawq_kernels', exact=True) }}
+        - packaging
+        - setuptools >=49.4.0
+        - pytorch
+        - wheel
+        - pip
+        - requests
+        - libxcrypt
+      run:
+        - python
+        - {{ pin_subpackage('autoawq_kernels', exact=True) }}
+        - accelerate
+        - datasets
+        - tokenizers >=0.12.1
+        - transformers >=4.35.0
+        - typing_extensions >=4.8
+        - zstandard
+    test:
+      imports:
+        - awq
```

If this is loading `libcuda.so` (which requires a working GPU) and it may be doing this, we would need to skip this test

```suggestion
      imports:  # [cuda_compiler_version != "None"]
        - awq   # [cuda_compiler_version != "None"]
```

Alternatively we could switch to a Quansight GPU runner

---

It fails to find `libtorch_python.so`:
```
import: 'awq_ext'
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/autoawq_1723203836924/test_tmp/run_test.py", line 2, in <module>
    import awq_ext
ImportError: libtorch_python.so: cannot open shared object file: No such file or directory
```
I don't understand how the build manages to link at buildtime, but not at runtime. Well, `-Wl,--allow-shlib-undefined` might explain some of it (didn't succeed in stripping this it seems), but `-L$SP_DIR/torch/lib` is in `LDFLAGS`, so I have no idea where this fails. I doubt it's the cuda drivers though.

---

IIUC this was an issue in PyTorch packaging that was being worked on in PR: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/246

AIUI this was added to PyTorch 2.4.0: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/250

So likely we need to update PyTorch here too